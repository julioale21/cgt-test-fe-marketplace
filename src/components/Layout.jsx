import { Outlet } from 'react-router-dom';
import { Header } from '../components/shared/Header';
import { Box } from '@mui/material';

export function Layout() {
  const cartItemsCount = 3;

  return (
    <Box sx={{ display: 'flex', flexDirection: 'column', minHeight: '100vh' }}>
      <Header cartItemsCount={cartItemsCount} />

      <Box component="main" sx={{ flexGrow: 1, p: 3 }}>
        <Outlet />
      </Box>

      <Box component="footer" sx={{ p: 2, textAlign: 'center', mt: 'auto' }}>
        <p>Â© 2025</p>
      </Box>
    </Box>
  );
}
