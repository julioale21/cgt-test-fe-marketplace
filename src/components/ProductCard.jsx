import React from 'react';

import { Link } from 'react-router-dom';

import { Button, Card, CardActions, CardContent, CardMedia, Grid, Typography } from '@mui/material';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';

const ProductCard = ({ product }) => {
  return (
    <Grid item xs={12} sm={6} md={4} lg={3}>
      <Card sx={{ height: '100%', display: 'flex', flexDirection: 'column' }}>
        <CardMedia
          component="img"
          height="200"
          image={product.image}
          alt={product.name}
          sx={{
            objectFit: 'cover',
            height: '200px'
          }}
        />
        <CardContent sx={{ flexGrow: 1 }}>
          <Typography gutterBottom variant="h6" component="h2">
            {product.name}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mb: 2 }}>
            {product.description}
          </Typography>
          <Typography variant="h6" color="primary" sx={{ mt: 2 }}>
            ${product.price.toFixed(2)}
          </Typography>
          <Typography variant="body2" color="text.secondary" sx={{ mt: 1 }}>
            Formats: {product.format.join(', ')}
          </Typography>
        </CardContent>
        <CardActions sx={{ justifyContent: 'space-between' }}>
          <Button component={Link} to={`/products/${product.id}`} size="small" color="primary">
            View Details
          </Button>
          <Button size="small" variant="contained" color="primary" startIcon={<ShoppingCartIcon />}>
            Add to Cart
          </Button>
        </CardActions>
      </Card>
    </Grid>
  );
};

export { ProductCard };
