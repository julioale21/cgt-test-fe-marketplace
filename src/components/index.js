export { ContainedButton } from "./buttons/ContainedButton";
export { Header } from "./shared/Header";
export { Layout } from "./Layout";
export { OutlinedButton } from "./buttons/OutlinedButton";
export { ProductList } from "./ProductList";
export {ProductCard} from "./ProductCard";
