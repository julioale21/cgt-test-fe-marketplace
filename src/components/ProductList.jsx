import React from 'react';
import { Grid } from '@mui/material';
import { ProductCard } from './ProductCard';

const ProductList = ({ products }) => {
  return (
    <Grid container spacing={4}>
      {products.map((product) => (
        <ProductCard key={product.id} product={product} />
      ))}
    </Grid>
  );
};

export { ProductList };
