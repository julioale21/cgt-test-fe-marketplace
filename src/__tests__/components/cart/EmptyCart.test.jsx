import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { EmptyCart } from '../../../components';

describe('EmptyCart', () => {
  const defaultProps = {
    onContinueShopping: jest.fn()
  };

  it('should match snapshot', () => {
    const props = { ...defaultProps };

    const { container } = render(<EmptyCart {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should display empty cart message', () => {
    const props = { ...defaultProps };

    render(<EmptyCart {...props} />);

    expect(screen.getByText('Your cart is empty')).toBeInTheDocument();
  });

  it('should render shopping cart icon', () => {
    const props = { ...defaultProps };

    render(<EmptyCart {...props} />);

    expect(screen.getByTestId('ShoppingCartIcon')).toBeInTheDocument();
  });

  it('should render continue shopping button with correct text', () => {
    const props = { ...defaultProps };

    render(<EmptyCart {...props} />);

    expect(screen.getByText('Start Shopping')).toBeInTheDocument();
  });

  it('should render back arrow icon in button', () => {
    const props = { ...defaultProps };

    render(<EmptyCart {...props} />);

    expect(screen.getByTestId('ArrowBackIcon')).toBeInTheDocument();
  });

  it('should call start shopping when button is clicked', async () => {
    const props = { ...defaultProps };
    const user = userEvent.setup();

    render(<EmptyCart {...props} />);
    await user.click(screen.getByText('Start Shopping'));

    expect(props.onContinueShopping).toHaveBeenCalledTimes(1);
  });

  it('should render card with correct max width', () => {
    const props = { ...defaultProps };

    render(<EmptyCart {...props} />);
    const card = screen.getByText('Your cart is empty').closest('.MuiCard-root');

    expect(card).toHaveStyle({
      maxWidth: '800px'
    });
  });
});
