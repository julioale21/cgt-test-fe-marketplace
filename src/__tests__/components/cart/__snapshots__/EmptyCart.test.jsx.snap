// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyCart should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-l03i02-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiCardContent-root css-19iapi7-MuiCardContent-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-dhyiz8-MuiSvgIcon-root"
        data-testid="ShoppingCartIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2M1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-2qvmir-MuiTypography-root"
      >
        Your cart is empty
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-eicl6u-MuiButtonBase-root-MuiButton-root"
        id=":r0:"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20z"
            />
          </svg>
        </span>
        Continue Shopping
      </button>
    </div>
  </div>
</div>
`;
